allow arbitrary user-data to be provided

gottaeat opened the pull request:
fixes #1

randomthingsandstuff commented on cloudvirt/cli.py:
```diff
@@ -46,26 +44,33 @@ def _nuke_args(self):
 
         create_subparser.add_argument(
-            "--vmspec",
+            "--vmcfg",
```
can this just be a mandatory positional argument?

randomthingsandstuff commented on cloudvirt/cloudinit.py:
```diff
@@ -43,15 +43,48 @@ def _gen_netconf(self):
 
-        cloudinit_udata["users"] = []
+        # init users key if no user-data
+        if not self.vmspec.userdata:
```
should be setup if userspec exists. so basically, we should not be checking if there is a user supplied userdata, but instead just filling that in the userdata dict. if there is no user supplied userdata, go ahead and make the userdata an empty dict. userspec will override any userdata stuff for a given user, if supplied.

randomthingsandstuff reviewed (dismissed):
plz see the comments in the review. hopefully it shoes up now (learning how to github better)

gottaeat committed:
upd: (cloudvirt/) spec.py; add VMSpec().userdata attrib, set VMSpec().sshpwauth None by default.

gottaeat committed:
upd: (cloudvirt/) config.py:
 > make ConfigYAML take userdata_file as an arg.
 > _parse_userspec(); make userspec_file not mandatory.
 > add _parse_userdata():
  - load user-data file if specified.
  - check if the users key exists, see if it has any users in it.
 > parse_yaml() -> run(), call _parse_userdata() too.

gottaeat committed:
upd: (cloudvirt/) cloudinit.py; rewrite _gen_udata() allow for merging userspec information with what's provided within user-data.

gottaeat committed:
upd: (cloudvirt/) cli.py:
> redo prompts.
> make --vmspec a positional argument.
> rename --userspec to --users, make it non-required.
> add --userdata flag, non-required.
> if neither userspec or users is provided, die because for almost all of the
  cloudimgs you will not have a password provided for the default user.

gottaeat committed:
upd: (/) .gitignore; ignore /user-data.

gottaeat committed:
upd: (tests/) test_cloudinit_gen.py; match current behavior of CloudInit().
